### ISSUE TITLE
Untangle CheckoutSessionMixin

### ISSUE THREAD

**Issue by maiksprenger** [django-oscar/django-oscar#1468]
I recently had to work on a checkout REST API based on Oscar, and had to get the shipping address and method stored in the session. The mixin currently carries the request and session data as state around and is initialised in `dispatch`. That makes sense when it's used in views, but there's a lot of methods in there that could be helpful outside of that context. 

I'm not entirely sure what the right approach is. Maybe limiting the scope of the mixin to handle the  validating the pre and skip conditions makes sense. All other logic could live in individual methods.

**Comment by sasha0** [comment]
Yes, it does make sense to decouple `CheckoutSessionMixin` into two classes:

- session validation mixin, with all methods which require request parameter and with the initiation in dispatch method;
- order submission data mixin.

Backwards compatibility would be supported by inheriting both mixin classes under the old name, but will make the whole functionality more re-usable.

**Comment by sasha0** [comment]
Made attempt to decouple, hence reviewed mixin again and now not sure if it does make sense, since checkout methods are interrelated, e.g.:

`check_a_valid_shipping_address_is_captured` calls `get_shipping_address`, `skip_unless_payment_is_required` calls `get_shipping_address` and `get_order_totals`.
